Automattic/jetpack
# Payments - modify link to open new payment plan modal

#### Changes proposed in this Pull Request:
- for payments blocks add hash "add-new-payment-plan" to link to new payment plan. This hash allow to open modal modal on page load.

Fixes # https://github.com/Automattic/wp-calypso/issues/62888

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
Checkout the PR
Login to admin panel
Choose paid plan to have access to payments blocks
Choose theme with widget areas (eg. twenty twenty one)
Go to Customizer
Go to Widgets, select any widget area
Add new payment block
Browse for current plans and click on the bottom to "Add a new payment plan"
Check if you are redirected to /earn/payment-plans#add-new-payment-plan, not just add-new-payment-plan

<img width="577" alt="link-hash" src="https://user-images.githubusercontent.com/82798599/172669530-21572f42-9ce6-4d8b-b794-be718cbcb7fe.png">
